webapp/src/DOMJudgeBundle/Service/EventLogService.php

```diff
@@ -414,18 +443,350 @@ public function log(string $type, $dataIds, string $action, $contestId = null, $
+            // Special case for state: this is always an update event
+            if ($endpointType === 'state') {
+                $event->setAction(self::ACTION_UPDATE);
+            } else {
+                // Set the action basd on whether there was already an event
+                // for the same endpoint and ID
+                $event->setAction(
+                    $existingEvent === null ?
+                        self::ACTION_CREATE :
+                        self::ACTION_UPDATE
+                );
+            }
+            $this->em->persist($event);
+            $this->em->flush();
+        }
+
+        // Make sure to release the lock again
+        if ($this->em->getConnection()->fetchColumn('SELECT RELEASE_LOCK(:lock)',
+                                                    [':lock' => $lockString]) != 1) {
+            throw new Exception('EventLogService::addMissingStateEvents failed to release lock');
+        }
+    }
+
+    /**
+     * Add all static events that are still missing. Static events are events for endpoints that
+     * are marked as 'Configuration' on
+     * https://clics.ecs.baylor.edu/index.php?title=Contest_API#Types_of_endpoints
+     *
+     * @param Contest $contest
+     * @throws NonUniqueResultException
+     * @throws Exception
+     */
+    public function initStaticEvents(Contest $contest)
+    {
+        // Loop over all configuration endpoints with an URL and check if we have all data
+        foreach ($this->apiEndpoints as $endpoint => $endpointData) {
+            if ($endpointData[EventLogService::KEY_TYPE] === EventLogService::TYPE_CONFIGURATION &&
+                isset($endpointData[EventLogService::KEY_URL])) {
+                $contestId = $contest->getApiId($this);
+
+                // Do an internal API request to the overview URL
+                // of the endpoint to get current data
+                $url = sprintf('/contests/%s%s', $contestId,
+                               $endpointData[EventLogService::KEY_URL]);
+                $this->dj->withAllRoles(function () use ($url, &$data) {
+                    $data = $this->dj->internalApiRequest($url);
+                });
+
+                // Get a partial reference to the contest,
+                // because calling internalApiRequest above will clear the entity manager
+                /** @var Contest $contest */
+                $contest = $this->em->getPartialReference(Contest::class, $contest->getCid());
+
+                if ($data === null) {
+                    throw new Exception(sprintf("EventLogService::initializeStaticEvents no response data for endpoint '%s'.",
+                                                $endpoint));
+                }
+
+                if (!is_array($data)) {
+                    throw new Exception(sprintf("EventLogService::initializeStaticEvents Endpoint '%s' did not return a JSON list.",
+                                                $endpoint));
+                }
```

Comment:
Isn't this in the wrong order with the `$endpoint === 'contests'` check below?

Reply:
Well Json objects are also php arrays...
This is literally what was there in the event daemon code this is based on

Reply:
I see, so that was unclear to begin with. I would still prefer to fix it to say "did not return a JSON object/array" to clarify that we're not just verifying that the endpoint data is a collection.

Reply:
Maybe even add an extra check in the non-contest loop to check that that are also arrays?